### ISSUE TITLE
Consider Firefox Klar for Android as alternative to Focus.

### ISSUE THREAD

**Issue by Atavic** [nylira/prism-break#1753]
[Fossdroid](https://fossdroid.com/) lists Firefox Klar:

> Firefox Klar is a version of Firefox Focus for Germany, Austria and Switzerland...

**EDIT**: https://github.com/nylira/prism-break/issues/1753#issuecomment-317110710

**Comment by strugee** [comment]
Please file a separate issue for Fossdroid.

**Comment by Zegnat** [comment]
Interesting that Klar would have Telemetry turned off.

I have grabbed Mozilla’s official built of both Focus and Klar from the [v1.0 release](https://github.com/mozilla-mobile/focus-android/releases/tag/v1.0) and can confirm that:

* Klar will work in non-German languages. It is set to “System default” by default and has no problem displaying in English on my phone.
* The setting “Send anonymous usage data” is off-by-default in Klar and on in Focus. So Klar would be preferable.

I am not sure if Focus/Klar should be considered a full browser though? Does it really replace e.g. Chrome in all cases?

* [Homepage Firefox Focus](https://www.mozilla.org/en-US/firefox/focus/)
* [Blog post introducing Firefox Focus](https://blog.mozilla.org/blog/2017/06/20/firefox-focus-new-to-android-blocks-annoying-ads-and-protects-your-privacy/)
* [Source on GitHub](https://github.com/mozilla-mobile/focus-android)

The link for the F-Droid build did not work for me on Fossdroid. So the only build currently available would be straight from [the GitHub releases](https://github.com/mozilla-mobile/focus-android/releases) (and from the Google Play Store).

**Comment by Atavic** [comment]
Cons: No tabs, no Webextenisons, no bookmarks and no data synchronization.

**Comment by Zegnat** [comment]
Wait what? Can you be more specific @Atavic? From [the (apparent) source article (German)](http://www.deutschlandfunk.de/anti-tracking-software-mozilla-klar-saugt-daten-ab.684.de.html?dram:article_id=378712):

> Wenn man als Anwender die Funktion für die Datenübermittlung abschaltet, kann man Firefox Klar verwenden, ansonsten Nein, Klares Nein.

Rough translation:

> When the user turns off the function for data transmissions, they can use Firefox Klar, otherwise no, big no.

And the function **is turned off**. That’s specifically what Klar does and Focus does not. By admission of the source you just linked Klar does not send data!

In fact, directly from the in-app documentation:

> Firefox Klar does not utilize Adjust. Mozilla’s telemetry system is turned off by default on Klar. You can turn telemetry on <i>[… bla bla bla.]</i>

Unless you have a source newer than one dated February, this really shouldn’t be used to close the topic. The Focus/Klar Telemetry difference seems to be [a thing from April](https://github.com/mozilla-mobile/focus-android/issues/411) and was obviously not testable in February.

**Comment by quakenet2** [comment]
No. You should stick to Orfox (or Tor Browser Android Edition which is WIP).

See these and how they respond.

https://github.com/mozilla-mobile/focus-android/issues/1115
https://github.com/mozilla-mobile/focus-android/issues/1116
https://github.com/mozilla-mobile/focus-android/issues/1117

**Comment by quakenet2** [comment]
Privacy <-----> Not Private

Orfox ------------ Klar ---------------- Focus -- FirefoxAndroid

**Comment by quakenet2** [comment]
Orfox ------------ Klar ---------------- Focus -- FirefoxAndroid--------------- ChromeRelated

**Comment by Zegnat** [comment]
@quakenet2, is there any data behind those issues you files against Klar or are they just questions?

**TL;DR:** Unless you have factual information regarding your concerns, I don’t think any of those are actionable in regards to PRISM Break.

---

1. At a cursory glance through the code I don’t see any of those URLs you mention. Are these things Firefox for Desktop connects to? And if so, why would Klar connect to them? Did you notice these in a connection log?

2. Yes, Klar also includes Telemetry, this doesn’t need a separate button because it goes together with all the other anonymous usage data. Which you can read about on the page that describes what anonymous usage data is. This makes me think this is FUD, unless you have reason to believe the application acts against its own documentation.

3. It doesn’t download the block lists at all, these are compiled into the application. You can [review them here](https://github.com/mozilla-mobile/focus-android/tree/master/shavar-prod-lists).

In the first issue, you write:

> If you implement a proxy support(http/socks), I can monitor your app’s activity.
> Until then, I can’t endorse your app as a “privacy app” even this is a open source.

I am not sure why you would need the application to implement something for you to do this? You can always set-up a global HTTP proxy for your Android device in your WiFi settings. Or just run proxy software on whatever you use as an access point. But why would you even go through that trouble when an application is open source and you can review what connections it makes right there?

**Comment by quakenet2** [comment]
@Zegnat just a question. If they add proxy support(and someone already asked for this issue, not me), I can start analyzing. I hate current Mozilla Firefox because they send too much by default without killswitch.

**Comment by quakenet2** [comment]
@Zegnat And since you're not a Mozilla developer, I'll just wait for their answer. Thank you.

**Comment by quakenet2** [comment]
https://github.com/mozilla-mobile/focus-android/issues/1119 - more details about "data sending"
https://github.com/mozilla-mobile/focus-android/issues/1120 - Mozilla Focus / Mozilla Klar's definition violate the law

**Comment by Zegnat** [comment]
@quakenet2: Unless you have **factual** and **new** information regarding **privacy features** in the Klar (not Focus) browser, please do not cross-post all your questions here.

This will be my last comment to things that aren’t new facts.

> If they add proxy support(and someone already asked for this issue, not me), I can start analyzing.

And I am wondering what is wrong with the OS wide proxy support you already have on Android. If you do not trust Mozilla’s code, you should not trust them to correctly implement proxies either and the only way to actually monitor their sending behaviour is with a proxy either on the OS or on your Access Point.

That’s why I didn’t understand how them not implementing proxy support in their app (an app you do not trust) is keeping you from monitoring the “app’s activity”.

> And since you're not a Mozilla developer, I'll just wait for their answer.

Again, you don’t trust their application but you do trust their developers? Even more than a third-party developer with full access to the code and who has [gone through it before](https://github.com/nylira/prism-break/issues/1753#issuecomment-317279768) to find the definitive different between Focus and Klar?

And surprise, they gave [the exact same answer](https://github.com/mozilla-mobile/focus-android/issues/1117#issuecomment-321482213) as [I did](https://github.com/nylira/prism-break/issues/1753#issuecomment-321467583) in regards to the block lists. Probably because we have read the same code.

(Glancing over whether you have ever seen my CV and know if I work for Mozilla or not.)

> mozilla-mobile/focus-android#1119 - more details about "data sending"

I don’t see anything new there that isn’t already described [on their wiki](https://github.com/mozilla-mobile/focus-android/wiki/Telemetry) or in [the official support documentation](https://support.mozilla.org/en-US/kb/send-anonymous-usage-data-firefox-mobile-devices) that the application itself links to.

Why should it even mention Health Reports? Or Crash Logs? Those do not exist for the mobile applications.

> mozilla-mobile/focus-android#1120 - Mozilla Focus / Mozilla Klar's definition violate the law

This links to a very old forum post that seems to rehash the points @Atavic correctly brought up before. These were also points I have addressed above already. What you are left with is whether the label “anonymous” is fine if it includes possible PID?

This is an interesting legal debate indeed, especially with GDPR being relatively unexplored. If you are worried about legal aspects I assume you know the new definitions introduced by GDPR in Europe? I’d love to debate these, but PRISM Break is not the place for that.

**Comment by Atavic** [comment]
>mozilla-mobile/focus-android#1115

>Are these things Firefox for Desktop connects to? 

Yes.

**Comment by Atavic** [comment]
@quakenet2 >mozilla-mobile/focus-android#1115

@Zegnat >Are these things Firefox for Desktop connects to?

Yes. The first one is related to the shield-recipe-client addon & goes to:
https://github.com/mozilla/normandy/tree/master/recipe-client-addon

You can deactivate under about:config

```user_pref("extensions.shield-recipe-client.enabled", false);
user_pref("extensions.shield-recipe-client.api_url", "");```

**Comment by Atavic** [comment]
@quakenet2 >mozilla-mobile/focus-android#1115

@Zegnat >Are these things Firefox for Desktop connects to?

Yes. The first one is related to the (shield-recipe-client addon)[https://github.com/mozilla/normandy/tree/master/recipe-client-addon].

You can deactivate under about:config

user_pref("extensions.shield-recipe-client.enabled", false);
user_pref("extensions.shield-recipe-client.api_url", "");

More info (here)[https://github.com/ghacksuserjs/ghacks-user.js/search?utf8=%E2%9C%93&q=Shield+Recipe+Client&type=].

**Comment by Atavic** [comment]
@quakenet2 >mozilla-mobile/focus-android#1115

@Zegnat >Are these things Firefox for Desktop connects to?

Yes. The first one is related to the [shield-recipe-client addon](https://github.com/mozilla/normandy/tree/master/recipe-client-addon).

You can deactivate under about:config

user_pref("extensions.shield-recipe-client.enabled", false);
user_pref("extensions.shield-recipe-client.api_url", "");

More info [here](https://github.com/ghacksuserjs/ghacks-user.js/search?utf8=%E2%9C%93&q=Shield+Recipe+Client&type=).

**Comment by Atavic** [comment]
>mozilla-mobile/focus-android#1115

>Are these things Firefox for Desktop connects to?

@Zegnat Yes. The first one is related to the [shield-recipe-client](https://github.com/mozilla/normandy/tree/master/recipe-client-addon) addon.

You can deactivate under about:config

user_pref("extensions.shield-recipe-client.enabled", false);
user_pref("extensions.shield-recipe-client.api_url", "");

More info [here](https://github.com/ghacksuserjs/ghacks-user.js/search?utf8=%E2%9C%93&q=Shield+Recipe+Client).

**Comment by Atavic** [comment]
>mozilla-mobile/focus-android#1115

>Are these things Firefox for Desktop connects to?

@Zegnat Yes. The first one is related to the [shield-recipe-client](https://github.com/mozilla/normandy/tree/master/recipe-client-addon) addon.

You can deactivate under about:config
More info [here](https://github.com/ghacksuserjs/ghacks-user.js/search?utf8=%E2%9C%93&q=Shield+Recipe+Client).

**Comment by Zegnat** [comment]
@Atavic, thanks for the confirmation. That means those URL calls have nothing to do with Klar then.

**Comment by Atavic** [comment]
Yes. It is a general issue: https://github.com/pyllyukko/user.js/issues/319

**Comment by quantumpacket** [comment]
Just something to consider https://github.com/mozilla-mobile/focus-android/issues/344

**Comment by Atavic** [comment]
@quantumpacket I looked at that. Like many other issues it has been labeled as low priority **P5** and likely ignored.
